src/main/scala/chisel3/tester/internal/Utils.scala

```diff
@@ -0,0 +1,29 @@
+// See LICENSE for license details.
+
+package chisel3.tester.internal
+
+import treadle.utils.BitMasks
+
+object Utils {
+
+  /** Converts an unsigned BigInt of width X to it's signed value
+    * Basically if msb is set convert it to a negative number
+    *
+    * @param unsigned an unsigned value that should be converted to signed
+    * @param width    the width of the target (this helps to determine if sign (MSB) is set
+    * @return         a signed BigInt
+    */
+  def unsignedBigIntToSigned(unsigned: BigInt, width: Int): BigInt = {
```

Comment:
I'm not sure if this is useful as a general util as opposed to a specific communication protocol detail for just the Verilator backend.

Reply:
Moved into verilator utils